### PR TITLE
Fix local cloning under Python 3

### PR DESCRIPTION
**Pull request by myint**
This gets local cloning working. It supports Python 3 while leaving the Python 2 effectively untouched.

```
$ python examples/clone.py . /tmp/clone
```

The above used to result in a `TypeError`. This did not show up in the tests since the tests don't use the default arguments.

### CODE REVIEW

**Comment by myint** [comment]
Note that the Travis CI errors are unrelated to this pull request.

File: dulwich/porcelain.py [hunk 1]
```diff
@@ -97,6 +98,14 @@
 GitStatus = namedtuple('GitStatus', 'staged unstaged untracked')
 
 
+if sys.version_info[0] < 3:
+    STDOUT = sys.stdout
```
Comment (jelmer) [inline dulwich/porcelain.py]: Please use a more descriptive name; these are not the same as stdout/stderr.

File: dulwich/porcelain.py [hunk 2]
```diff
@@ -97,6 +98,14 @@
...
+    STDERR = sys.stderr
+else:
+    STDOUT = codecs.getwriter('utf-8')(sys.stdout.buffer)
```
Comment (jelmer) [inline dulwich/porcelain.py]: The system encoding is not necessarily utf-8. Is there any reason why just sys.stdout.buffer wouldn't work?

**Comment by myint** [comment]
I've made the relevant changes.

**Comment by jelmer** [comment]
Thanks, merged with two changes:

clarified variables names to not be capitalized (as they're not constants) and include the word 'bytes'.
I've only merged the change for clone() as that's the only one we've verified is correct. For some of the other functions the outstream is not actually a bytestring (e.g. print_commit). This needs adjusting/renaming.

**Pull request closed without merging**